src/main/java/seedu/meetingjio/commands/ListCommand.java

```diff
@@ -18,7 +25,32 @@
      */
     @Override
     public String execute(MasterTimetable masterTimetable) {
-        Timetable timetable = masterTimetable.getByIndex(0); // changes needed
+        String user = this.name;
+        if (user.equals("all")) {
+            return listAll();
+        }
+        return listUser(user);
+    }
+
+    private String listAll() {
+        String str = "";
+        for (Timetable timetable : timetables) {
+            String user = timetable.getName();
+            str += user;
+            str += '\n';
+            str += listUser(user);
+            str += '\n';
+        }
+        return str.substring(0, str.length() - 1); //remove last newline character
+    }
+
+    private String listUser(String user) {
+        Timetable timetable = null;
```

Comment:
Can consider get the timetable using getByName method in MasterTimetable.
And the exception thrown can be handled in the execute function.

Reply:
done!